### PR TITLE
Modify S2823: LaYC format

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S2823/python/rule.adoc [hunk 1]
```diff
@@ -1,24 +1,31 @@
+This rule raises an issue when the ``++__all__++`` property of a module contains objects that aren't strings.
+
 == Why is this an issue?
 
-``++__all__++`` is used to define the list of module's names that should be imported when ``++from package import *++`` is used. For that reason, it may only contain strings.
+``++__all__++`` is used to define the list of module's names that should be imported when ``++from package import *++`` is used. For that reason, it may only contain strings. If something else than a string is listed, a `TypeError` will be raised when trying to perform a wildcard import of the module.
+
+== How to fix it
```
Comment (joke1196) [inline rules/S2823/python/rule.adoc:7]: I think this one may deserve only one tab or, we could add an example of import with a file exporting with `__all__`

**Review by joke1196 — APPROVED** [review]
LGTM! I have added a tiny comment, let me know what you think

File: rules/S2823/python/rule.adoc (hunk 1 above)
Comment (jean-jimbo-sonarsource) [reply rules/S2823/python/rule.adoc]: A grammatical correction "...used to define the list of _**module**_ names... If something _**other**_ than a string is listed..."

Comment (guillaume-dequenne-sonarsource) [reply rules/S2823/python/rule.adoc]: Sorry, I applied the fix a bit hastily. It's not actually correct: it's not a list of module names, it's the list of names from that module.
I reworded again the whole sentence. Wdyt?

**Review by jean-jimbo-sonarsource — APPROVED** [review]
LGTM

**Review by joke1196 — APPROVED** [review]
LGTM!

**Pull request merged**